fix for #2110 broke compilation on some platforms due to INT_MAX

This looks somewhat like https://github.com/libarchive/libarchive/issues/627

Commit https://github.com/libarchive/libarchive/commit/d9f44c5b44038c735a78cc1b32fda1ea7b88be25 introduced usage of INT_MAX

This produces the following failure:
```
/gentoo/prefix64/var/tmp/portage/app-arch/libarchive-3.7.4/work/libarchive-3.7. 4/tar/bsdtar.c: In function ‘main’:
/gentoo/prefix64/var/tmp/portage/app-arch/libarchive-3.7.4/work/libarchive-3.7. 4/tar/bsdtar.c:374:43: error: ‘INT_MAX’ undeclared (first use in this function)
  374 |                         if (l < 0 || l >= INT_MAX || *(bsdtar-          >argument) == '\0' ||
      |                                           ^~~~~~~
/gentoo/prefix64/var/tmp/portage/app-arch/libarchive-3.7.4/work/libarchive-3.7. 4/tar/bsdtar.c:71:1: note: ‘INT_MAX’ is defined in header ‘<limits.h>’; did you forget to ‘#include <limits.h>’?
   70 | #include "err.h"
  +++ |+#include <limits.h>
   71 |
```

My env:
```
% uname -a
SunOS hollandscheleeuw 5.11 illumos-a6e309ba23 i86pc i386 i86pc Solaris
% cat /etc/release 
             OpenIndiana Hipster 2023.10 (powered by illumos)
        OpenIndiana Project, part of The Illumos Foundation (C) 2010-2023
                        Use is subject to license terms.
                           Assembled 20 October 2023
% gcc --version
gcc (Gentoo 13.2.0-r1 p7) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/tar/bsdtar.c
+++ b/tar/bsdtar.c
@@ -43,6 +43,9 @@
 #ifdef HAVE_LANGINFO_H
 #include <langinfo.h>
 #endif
+#ifdef HAVE_LIMITS_H
+#include <limits.h>
+#endif
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif

